v2/sender/sender.go

```diff
@@ -0,0 +1,83 @@
+package sender
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
+)
+
+// MessageBody is a type to represent that an input message body can be of any type
+type MessageBody any
+
+// SendHandlerFunction is a function that is used to modify the ServiceBus message before sending
+type SendHandlerFunction func(ctx context.Context, msg *azservicebus.Message)
+
+// CustomSender is a wrapper around the ServiceBus sender that allows for users to introduce middleware to modify the ServiceBus message before it's sent
+type CustomSender interface {
+	SendMessage(ctx context.Context, mb MessageBody, options *azservicebus.SendMessageOptions, middleware SendHandlerFunction) error
+}
+
+// Sender is an interface that allows users to substitute their own senders for the default ServiceBus sender when making a CustomSender
```

Comment:
// Sender is satisfied by *azservicebus.Sender

Reply:
maybe call this SBSender to not conflict with goshuttle.Sender?

Reply:
renamed the field to be sbSender as well